#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation 0.5633470481966442 0.5693323050806764 0.5987493880411501 4.114997258572609
  position -0.4054009480325943 1.350767531033346 0.6195304674724398
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Robot {
  translation -0.0539257 0.0485206 0.239573
  rotation 0.0018680552589844 0.9999963882139613 0.0019323375952818218 -1.5721038512920498
  children [
    DistanceSensor {
      translation -0.015 -0.02 0.11
      rotation 0.5773372688908542 -0.5773372688908542 -0.5773762689090183 2.09448
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "inf_right"
      type "infra-red"
    }
    DistanceSensor {
      translation 0 -0.02 0.11
      rotation 0.5773372688908542 -0.5773372688908542 -0.5773762689090183 2.09448
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "inf_cent"
      type "infra-red"
    }
    DistanceSensor {
      translation 0.015 -0.02 0.11
      rotation 0.5773372688908542 -0.5773372688908542 -0.5773762689090183 2.09448
      children [
        Shape {
          appearance PBRAppearance {
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "inf_left"
      type "infra-red"
    }
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2768.9508427964515
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.06 0 0.05
        rotation 0.6355319259111518 -0.6355342603545204 -0.4384009295876265 3.9679227150160203
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 1
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2323.5518724354147
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation 0.06 0 -0.05
        rotation -0.49886471088273715 0.49886654331848135 0.7087073953285324 1.9085041596820589
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 2402.11526781694
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF WHEEL4 Solid {
        translation -0.06 0 -0.05
        rotation 0.6361317701523701 -0.6361341067990992 0.4366574964084829 2.3181892031248217
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 1956.0244093876533
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation -0.06 0 0.05
        rotation -0.6381028774613375 0.6381052213483658 -0.43086708421915254 3.9552491966657177
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    Camera {
      translation 0 0.01 0.1
      rotation 3.4339994638889944e-07 -0.9659258492010602 -0.2588189595935602 -3.1415853071795863
      children [
        Transform {
          children [
            Shape {
              appearance PBRAppearance {
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
        }
      ]
      fieldOfView 0.9
      width 256
      height 128
      antiAliasing TRUE
      recognition Recognition {
        frameColor 1 1 0.498039
      }
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "controller"
}
DEF floor Solid {
  translation -0.37 0 0.25
  children [
    Shape {
      appearance PBRAppearance {
        roughness 1
        metalness 0
      }
      geometry Box {
        size 10 0.01 10
      }
    }
  ]
  name "floor"
  boundingObject Box {
    size 10 0.01 10
  }
}
DEF line Solid {
  translation -0.2 0.0005 0.24
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line"
}
DEF line Solid {
  translation -0.377941 0.0005 0.285
  rotation 0 1 0 0.523599
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(1)"
}
DEF line Solid {
  translation -0.453817 0.0001 0.424203
  rotation 0 -1 0 -1.5708053071795867
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(2)"
}
DEF line Solid {
  translation -0.402837 0.0001 0.602502
  rotation 0 1 0 2.0944
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(3)"
}
DEF line Solid {
  translation -0.273478 0.0001 0.699277
  rotation 0 1 0 2.8798
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(4)"
}
DEF line Solid {
  translation -0.183478 0.0001 0.819277
  rotation 0 -1 0 -1.5708053071795867
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(5)"
}
DEF line Solid {
  translation -0.279154 0.0001 0.933278
  rotation 0 1 0 0.261813
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(6)"
}
DEF line Solid {
  translation -0.413641 0.0001 0.866498
  rotation 0 1 0 -1.0471853071795865
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(7)"
}
DEF line Solid {
  translation -0.541583 0.0001 0.739537
  rotation 0 1 0 -0.5235853071795864
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(8)"
}
DEF line Solid {
  translation -0.624523 0.0001 0.599278
  rotation 0 -1 0 1.57079
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(9)"
}
DEF line Solid {
  translation -0.624523 0.0001 0.409278
  rotation 0 -1 0 1.57079
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.01 0.02
      }
      castShadows FALSE
    }
  ]
  name "line(10)"
}
